### PR TITLE
Feature/bphh 176/email notifications

### PR DESCRIPTION
**Pull request by bkmaibach**
## Description

add emails and logic for:

submitter welcome
reviewer onboarding
application submitted (reviewer and submitter)
application updated
weekly reviewer reminder

ALSO all of the devise templates now use our layout

modify all templates to use the new layout, use dynamic values including URLs from the new frontend URL helper.

## What type of PR is this? (check all applicable)

- [ x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## Related Tickets & Documents

https://hous-bssb.atlassian.net/browse/BPHH-176

### CODE REVIEW

File: app/views/permit_hub_mailer/02a_notify_reviewer_application_received_email.html.erb [hunk 1]
```diff
@@ -170,7 +170,7 @@ table[class="body"] .article {
                         
 
                         <ol style="font-family: 'Open Sans', sans-serif; font-size: 16px; line-height: 26px; font-weight: normal; margin: 0; margin-bottom: 24px; padding-left: 24px;">
-                          <li style="list-style-position: initial; margin-left: 5px;">Log into the <a href="https://buildingpermit.gov.bc.ca" style="color: #1a5a96; text-decoration: underline;">Building Permit Hub</a></li>
+                          <li style="list-style-position: initial; margin-left: 5px;">Log into the <a href="<%= @root_url %>" style="color: #1a5a96; text-decoration: underline;">Building Permit Hub</a></li>
```
Comment (taw27) [inline app/views/permit_hub_mailer/02a_notify_reviewer_application_received_email.html.erb:173]: should this link be `"<%= FrontendUrlHelper.frontend_url("/login")` same as the link below or is this intended?

File: app/views/permit_hub_mailer/09_announce_maintenance_email.html.erb [hunk 2]
```diff
@@ -1,215 +1,41 @@
...
+          </p>
+          <p style="font-family: 'Open Sans', sans-serif; font-size: 16px; line-height: 26px; font-weight: normal; margin: 0; margin-bottom: 24px;">
+            During the maintenance period, there may be temporary
+            interruptions in system access or reduced
+            functionality. Please save any permits in progress
+            before the scheduled maintenance window.  We apologize
+            for any inconvenience this may cause and appreciate
+            your understanding as we work to improve our services.
+          </p>
+          <p style="font-family: 'Open Sans', sans-serif; font-size: 16px; line-height: 26px; font-weight: normal; margin: 0; margin-bottom: 24px;">
```
Comment (taw27) [inline app/views/permit_hub_mailer/09_announce_maintenance_email.html.erb:25]: Just confirming if this page is meant to be stubbed as there is some @TODOs

File: app/views/permit_hub_mailer/notify_application_updated.html.erb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+              </td>
+            </tr>
+          </table>
+          
+          <h2 style="color: #292929; font-family: 'Open Sans', sans-serif; font-weight: 700; line-height: 1.4; margin-top: 24px; margin-bottom: 16px; text-align: left; font-size: 22px;">Next steps:</h2>
+
+          <ul style="font-family: 'Open Sans', sans-serif; font-size: 16px; line-height: 26px; font-weight: normal; margin: 0; margin-bottom: 24px; padding-left: 24px;">
+            <li style="list-style-position: initial; margin-left: 5px;">Local Jurisdiction will contact you with further details.</li>
+            <li style="list-style-position: initial; margin-left: 5px;">Please contact them if you do not hear back within a estimate of 3 weeks.</li>
+          </ul>
```
Comment (taw27) [inline app/views/permit_hub_mailer/notify_application_updated.html.erb:47]: Should this be hard code or is this outside the scope for initial release?

File: app/views/permit_hub_mailer/notify_reviewer_application_received.html.erb [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+              </td>
+            </tr>
+          </table>
+          
+
+          <h2 style="color: #292929; font-family: 'Open Sans', sans-serif; font-weight: 700; line-height: 1.4; margin-top: 24px; margin-bottom: 16px; text-align: left; font-size: 22px;">Review the application:</h2>
+          
+
+          <ol style="font-family: 'Open Sans', sans-serif; font-size: 16px; line-height: 26px; font-weight: normal; margin: 0; margin-bottom: 24px; padding-left: 24px;">
+            <li style="list-style-position: initial; margin-left: 5px;">Log into the <a href="<%= root_url %>" style="color: #1a5a96; text-decoration: underline;">Building Permit Hub</a></li>
```
Comment (taw27) [inline app/views/permit_hub_mailer/notify_reviewer_application_received.html.erb:43]: should this link directly to the log in page or the root?

File: app/views/permit_hub_mailer/postdrop_io/_postdrop template generic html.txt [hunk 5]
```diff
@@ -117,7 +117,7 @@
...
                                 
-                                Application update description TODO text here Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  and a <a href="https://buildingpermit.gov.bc.ca/">@TODO link</a>.
+                                Application update description TODO text here Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  and a <a href="<%= root_url %>">@TODO link</a>.
```
Comment (taw27) [inline app/views/permit_hub_mailer/postdrop_io/_postdrop template generic html.txt:120]: @TODO intentional?

File: app/views/permit_hub_mailer/postdrop_io/email content preprocess/01 welcome.txt [hunk 6]
```diff
@@ -108,7 +108,7 @@
                           <tr>
                             <td class="basic-table-content border-top">
                               <p>
-                                To update your information or login preferences please login into the <a href="https://buildingpermit.gov.bc.ca">Building Permit Hub</a>.
+                                To update your information or login preferences please login into the <a href="<%= root_url %>">Building Permit Hub</a>.
```
Comment (taw27) [inline app/views/permit_hub_mailer/postdrop_io/email content preprocess/01 welcome.txt:111]: should this be the root_url or the login url?

**Review by taw27 — APPROVED** [review]

File: app/views/permit_hub_mailer/09_announce_maintenance_email.html.erb (hunk 2 above)
Comment (bkmaibach) [reply app/views/permit_hub_mailer/09_announce_maintenance_email.html.erb:25]: These are for future features to be completed later.

File: app/views/permit_hub_mailer/notify_application_updated.html.erb (hunk 3 above)
Comment (bkmaibach) [reply app/views/permit_hub_mailer/notify_application_updated.html.erb:47]: this was given to me hard coded

File: app/views/permit_hub_mailer/postdrop_io/_postdrop template generic html.txt (hunk 5 above)
Comment (bkmaibach) [reply app/views/permit_hub_mailer/postdrop_io/_postdrop template generic html.txt:120]: im not sure what this postdrop stuff is

**Pull request merged**